### ISSUE TITLE
Oversampling + image processing enlarges output image unnecessarily

### ISSUE THREAD

**Issue by jbarlow83** [jbarlow83/OCRmyPDF#472]
--oversample 400 --deskew for example will increase the size of output images.

It's always worked this way, but this isn't strictly necessary to enlarge output images. 

It would make sense to run the image processing with oversampling too, so those tools can work on the oversampled images.

We could reduce images back to their input DPI after showing them to Tesseract.

**Comment by jfilter** [comment]
Hey, sorry to hijack this but I have a very related question. So right now ``--oversample 10 --force-ocr`` should create a PDF where the dpi of the images (in the PDF) are 10, right?

**Comment by jbarlow83** [comment]
`--oversample 10` would ensure images have a DPI of at least 10. (Where the DPI is calculated based on how the image is positioned on the canvas, not whatever DPI is claimed in the image's metadata.)

It would be very rare for an image to have only 10 DPI. Thus `--oversample 10` probably has no effect.

**Comment by jfilter** [comment]
Okay, thanks! I thought I could force the DPI via `--oversample`.

Maybe you could make `--image-dpi` work for the PDFs? So this would set the resulting DPI (and also fix this issue).

**Comment by jbarlow83** [comment]
No, I just need to have the code make some decision about whether it needs to save the oversampled image.

You could use a higher optimization setting to get smaller files too. Oversampling + aggressive JPEG optimization is equivalent to downsampling, to a degree.
